compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirFunctionParameterChecker.kt

```diff
@@ -70,4 +78,46 @@ object FirFunctionParameterChecker : FirFunctionChecker() {
             }
         }
     }
+
+    private fun checkUninitializedParameter(function: FirFunction<*>, context: CheckerContext, reporter: DiagnosticReporter) {
+        // Corner case: fun foo(`_` : ... = `_`)
+        val parameterNames = function.valueParameters.map { it.name }
+        if (parameterNames.none { it.isEscapedUnderscore }) return
+
+        for ((index, parameter) in function.valueParameters.withIndex()) {
+            // If a parameter's default value refers to another parameter that appears later, it would be unresolved.
+            // Such case is handled in [ErrorNodeDiagnosticCollectorComponent#visitErrorNamedReference].
+            if (parameter.defaultValue == null || parameter.defaultValue?.hasUnresolvedNameError() == true) continue
+
+            // Alas, CheckerContext.qualifiedAccesses stack is not available at this point.
+            // Thus, manually visit default value expression and collect qualified accesses of interest.
+            val qualifiedAccesses: MutableMap<FirQualifiedAccessExpression, FirValueParameter> = mutableMapOf()
+            parameter.defaultValue?.accept(object : FirVisitorVoid() {
+                override fun visitElement(element: FirElement) {
+                    element.acceptChildren(this)
+                }
+
+                override fun visitQualifiedAccessExpression(qualifiedAccessExpression: FirQualifiedAccessExpression) {
+                    val namedReference = qualifiedAccessExpression.calleeReference as? FirResolvedNamedReference ?: return
+                    if (!namedReference.name.isEscapedUnderscore) return
+                    val referredParameterIndex = parameterNames.indexOf(namedReference.name)
+                    if (index <= referredParameterIndex) {
+                        qualifiedAccesses[qualifiedAccessExpression] = function.valueParameters[referredParameterIndex]
```

Comment:
Why not just report it here? Is there an advantage of collecting them in a map and reporting later?

Reply:
🤦